pkg/util/network.go

```diff
@@ -29,20 +28,18 @@ func DownloadFile(destPath string, fileURL string, progressBar bool, shaSumURL s
 	client.RetryMax = 4
 	client.RetryWaitMin = 500 * time.Millisecond
 	client.RetryWaitMax = 5 * time.Second
-	client.CheckRetry = func(ctx context.Context, resp *http.Response, err error) (bool, error) {
-		// Default retry policy only retries on
-		// - connection reset
-		// - connection refused
-		// - No Response
-		// - net.Error with Temporary() == true
-		if err != nil && strings.Contains(err.Error(), "context deadline exceeded") {
-			return true, nil
-		}
-		return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
-	}
+	// Default retry policy retries on:
+	// - connection reset
+	// - connection refused
+	// - No Response
+	// - net.Error with Temporary() == true
+	// "context deadline exceeded" cannot be retried until you run the request again
+	// See https://github.com/hashicorp/go-retryablehttp/issues/167
+	client.CheckRetry = retryablehttp.DefaultRetryPolicy
 	client.Backoff = retryablehttp.DefaultBackoff
 	client.Logger = nil
-	client.HTTPClient.Timeout = 5 * time.Minute
+	// Timeout for the entire request
+	client.HTTPClient.Timeout = 30 * time.Minute
```

Comment:
To test "context deadline timeout", set this timeout to very low value, restrict your download speed and:

```
rm -f ~/.ddev/bin/docker-compose
ddev version
```